### DISCUSSION TITLE
JsonRPC Block by uuid

### DISCUSSION THREAD

**Discussion by newhinton** [AsamK/signal-cli#1648] [Q&A] [answered]
Hi! 

I am writing a signal bot, and i am using the jsonrpc as my method to connect to signals services with great success.

Quick question: how do i block users? There is: https://github.com/AsamK/signal-cli/blob/master/man/signal-cli.1.adoc#block which generally seems to work, but for some accounts i do not have a number, but only a uuid. How would i block a user in that case? Thanks!

**Comment by AsamK** [comment, marked as answer]
See this post in the man page, you can just use the uuid instead of the phone number:

> In all places where a RECIPIENT argument is expected in signal-cli's interface, one of the following can be used:
- The phone number
- The ACI, just a UUID
- The PNI, just a UUID, with "PNI:" prefix
- The username, with "u:" prefix, e.g. "u:test.000"

**Reply by newhinton** [reply]
Ahh. It seems i had an issue with properly setting the RECIPIENT argument in json:


```
        val array = JSONArray()
        array.put(contactToBlock)
        map["recipient"] = array.toString()
```

did not work. Correctly, it has to be the JSONArray directly:


```
        val array = JSONArray()
        array.put(contactToBlock)
        map["recipient"] = array
```

Also, a note for anyone having issues with  single argument RECIPIENTS:
Do not pass an array, pass the number/id directly, otherwise it wont be recognized properly.

**Comment by pepa65** [comment]
When I try `signal-cli -a "u:NAME.NUM" receive` I get: `Invalid account (phone number), make sure you include the country code.`
